feat(nodejs): Add `expected_version` variable (#5081)
starship/starship: docs/config/README.md

### Variables

| Variable         | Example    | Description                                                              |
| ---------------- | ---------- | ------------------------------------------------------------------------ |
| version          | `v13.12.0` | The version of `node`                                                    |

> | expected_version | `>=12.0.0` | Optional `node` version as set in the engines property of `package.json` |

Reviewer: Please properly document when this is shown and it is not.

> | engines_version | `>=12.0.0` | `node` version as set in the engines property of `package.json`. Will show if installed node version does not match with the set engines value, is otherwise invisible. |

Reviewer: ```suggestion
| engines_version | `>=12.0.0` | `node` version requirement as set in the engines property of `package.json`. Will only show if the version requirement does not match the `node` version. |
```

Author: Got it, committing this!
